multigeiger/speaker.cpp

```diff
@@ -68,7 +68,7 @@ void IRAM_ATTR isr_audio() {
   }
   portEXIT_CRITICAL_ISR(&mux_audio);
 
-  if (!isr_sequence)
+  if (!p)
```

Comment:
p is **not** pointing "neverwhere" and it **is** defined, it is pointing to the integer right after the allocated tick_sequence array.

Accessing that location (== executing a `*p` operation) would be dirty on esp32, but due to lacking memory protection it would likely not cause a real cpu level issue. On another CPU with MMU, `*p` might cause a cpu exception I guess.

But even that is **irrelevant** when looking at the proposed code: there is no `*p`, just `!p`, which checks whether p is NULL (but does not access the memory p is pointing at).

As p is pointing to the place after the array, it is not NULL in your scenario and the function will continue to execute lines 74... - the pointer p is never used again in the remainder of the function.

Also, as duration_ms is 0 in your scenario, it will cleanly set up variables for termination in 105..118.

Next time the isr is invoked, there is nothing to do... (and there is also no issue with p, as it is a local dynamic variable and it gets re-assigned from isr_sequence, which is NULL by then).

Reply:
To avoid future issues, I guess we could add something like this above line 74:
``
// do not access *p below here!
``

Reply:
Alright, understood, thanks for the explanation. That was bugging me, but as I've also read additionally a bit more about NULL / nullptr and I am a bit more relaxed now.

Reply:
added the comment to avoid future issues.